d2/src/test/java/com/linkedin/d2/xds/IPv6AwarePickFirstLoadBalancerTest.java

```diff
@@ -0,0 +1,70 @@
+package com.linkedin.d2.xds;
+
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.LoadBalancer;
+import io.grpc.LoadBalancerRegistry;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.testng.annotations.Test;
+
+import static com.linkedin.d2.xds.IPv6AwarePickFirstLoadBalancer.*;
+import static org.mockito.Mockito.verify;
+import static org.testng.Assert.*;
+
+public class IPv6AwarePickFirstLoadBalancerTest
```

Comment:
Should we explicitly test a case of `contiguous block of ipv4s followed by contiguous block of ipv6s` to ensure the LB behaves right in that scenario?

Reply:
I think this is functionally the same, in fact it's more thorough. This test that it handles *any* interleaving.